spacy/kb/kb.pyx

```diff
@@ -0,0 +1,108 @@
+# cython: infer_types=True, profile=True
+
+from pathlib import Path
+from typing import Iterable, Tuple, Union
+from cymem.cymem cimport Pool
+
+from .candidate import Candidate
+from ..tokens import Span
+from ..util import SimpleFrozenList
+from ..errors import Errors
+
+
+cdef class KnowledgeBase:
+    """A `KnowledgeBase` instance stores unique identifiers for entities and their textual aliases,
+    to support entity linking of named entities to real-world concepts.
+    This is an abstract class and requires its operations to be implemented.
+
+    DOCS: https://spacy.io/api/kb
+    """
+
+    def __init__(self, vocab: Vocab, entity_vector_length: int):
+        """Create a KnowledgeBase."""
+        # Make sure abstract KB is not instantiated.
+        if self.__class__ == KnowledgeBase:
+            raise TypeError(
+                Errors.E1046.format(cls_name=self.__class__.__name__)
+            )
```

Comment:
This will break the `nel_emerson` [project](https://github.com/explosion/projects/tree/v3/tutorials/nel_emerson) :(

Reply:
We could drop the check, but `nel_emerson` - and anything else instantiating `KnowledgeBase` directly - will break anyway at a later point due to non-implemented functions.

Reply:
Oh yea, I know. I wasn't suggesting to drop the check. Just mentioning this that we need to be aware that this is still a breaking change, that we want to move this to `develop` and ensure we update the impacted projects accordingly when this gets released.

Reply:
I'll change the target branch to `develop` then?

Reply:
Yep, please do. But before you do, you might want to make sure `develop` is sync'ed with the latest from `master` (if not, create a PR) because otherwise we'll likely see some git history mess ;p